### PR TITLE
Data UI adapts to user rights

### LINKED ISSUE #98: Data Management UI adapts to user rights

### PR DESCRIPTION
**Pull request by xbill82**
Fixes #98

### CODE REVIEW

File: src/components/Common/CrudlDocument.vue [hunk 1]
```diff
@@ -43,7 +43,12 @@
             Toggle all
           </button>
 
-          <button class="btn btn-small waves-effect waves-light margin-right-5 primary" @click.prevent="create"><i class="fa fa-plus-circle left"></i>Create</button>
+          <button class="btn btn-small waves-effect waves-light margin-right-5 primary"
+                  @click.prevent="create",
+                  v-if="displayCreate">
```
Comment (stafyniaksacha) [inline src/components/Common/CrudlDocument.vue]: We have not to hide element, but disable it with a help text.

See exemple here: [src/components/Data/Indexes/List.vue#L19](https://github.com/kuzzleio/kuzzle-backoffice/blob/master/src/components/Data/Indexes/List.vue#L19)

File: src/components/Data/Documents/List.vue [hunk 2]
```diff
@@ -90,13 +111,23 @@
           return false
         }
         return this.indexesAndCollections[this.index].realtime.indexOf(this.collection) !== -1
+      },
+      rightsOnDocument () {
```
Comment (stafyniaksacha) [inline src/components/Data/Documents/List.vue]: This method should be in the `userAuthorization` service

File: src/components/Common/List.vue [hunk 3]
```diff
@@ -14,14 +14,19 @@
...
 
         <div class="collection">
           <div class="collection-item" transition="collection" v-for="document in documents">
-            <component :is="itemName" @checkbox-click="toggleSelectDocuments" :document="document"
-                       :is-checked="isChecked(document.id)"></component>
+            <component :is="itemName"
+                       @checkbox-click="toggleSelectDocuments"
+                       :document="document"
+                       :is-checked="isChecked(document.id)"
+                       :rights="rightsOnDocument">
```
Comment (stafyniaksacha) [inline src/components/Common/List.vue]: I don't understand why you are passing this in props ?

Each child component must handle their own policies.
You can directly call the `userAuthorization` service in the child component.

File: src/components/Data/Documents/DocumentItem.vue [hunk 4]
```diff
@@ -14,8 +14,12 @@
     <label class="item-title"><a @click="toggleCollapse">{{document.id}}</a></label>
 
     <div class="right actions">
-      <a v-link="{name: 'DataUpdateDocument', params: {id: document.id}}"><i class="fa fa-pencil"></i></a>
-      <dropdown :id="document.id" class="icon-black">
+      <a
+        v-link="{name: 'DataUpdateDocument', params: {id: document.id}}"
+        v-if="rights.canEdit">
```
Comment (stafyniaksacha) [inline src/components/Data/Documents/DocumentItem.vue]: Same as above: 
We have not to hide element, but disable it with a help text.

See exemple here: [src/components/Data/Indexes/List.vue#L19](https://github.com/kuzzleio/kuzzle-backoffice/blob/master/src/components/Data/Indexes/List.vue#L19)

File: src/components/Data/Documents/List.vue [hunk 5]
```diff
@@ -7,11 +7,31 @@
...
+      <div class="row valign-bottom empty-set empty-set-condensed">
+        <div class="col s1 offset-s1">
+          <i class="fa fa-6x fa-lock grey-text text-lighten-1" aria-hidden="true"></i>
+        </div>
+        <div class="col s10">
+          <p>
+            You are not allowed to list documents in collection <strong>{{collection}}</strong><br>
+          </p>
+          <p>
+            <em>Learn more about security & permissions on <a href="http://kuzzle.io/guide/#permissions" target="_blank">http://kuzzle.io/guide</a></em>
```
Comment (stafyniaksacha) [inline src/components/Data/Documents/List.vue]: please replace `&` by  `&amp;` to have a valid document

File: src/components/Data/Documents/List.vue [hunk 6]
```diff
@@ -40,7 +60,8 @@
               Here you'll see the documents in <strong>{{collection}}</strong> <br/>
               <em>Currently there is no document in this collection.</em>
             </p>
-            <button v-link="{name: 'DataCreateDocument', params: {index: index, collection: collection}}"
+            <button v-if="canCreateDocument(index, collection)"
```
Comment (stafyniaksacha) [inline src/components/Data/Documents/List.vue]: Same as above: 
We have not to hide element, but disable it with a help text.

See exemple here: [src/components/Data/Indexes/List.vue#L19](https://github.com/kuzzleio/kuzzle-backoffice/blob/master/src/components/Data/Indexes/List.vue#L19)

**Comment by AnthonySendra** [comment]
+1

**Pull request merged**